docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -168,4 +176,22 @@ test('My test', async t => {
         .useRole(role);
 ```
 
-**Default value**: `false`
\ No newline at end of file
+**Default value**: `false`
+
+## Troubleshooting
+
+### Local Storage Values Set in the Role Code Are Lost
+
+Role's initialization actions start with clear local and session storages. Role actions may add or modify items in these storages.
+
+When navigation occurs in the role code, the destination page is opened with empty storages. To preserve the storage content, TestCafe saves it to the server before the navigation, and then restores on the destination page. However, if the browser retrieves the destination page from cache, TestCafe is unable to restore the storages.
```

Comment:
When the first navigation occurs in the role code.... To apply the changes of the local and session storages for the next visited pages, TestCafe saves it to the server...